middleware/multer.js

```diff
@@ -0,0 +1,44 @@
+const multer = require('multer');
+const { uploadPath } = require('../config');
+const { UnprocessableEntity } = require('../utils/customError');
+const { VALID_IMAGE_FORMATS } = require('../constant');
+const expectedMimeTypes = {
+  'image/jpeg': true,
+  'image/png': true,
+  'image/gif': true,
+  'image/svg': true,
+  'image/tiff': true,
+  'image/webp': true,
+};
+
+const storage = multer.diskStorage({
+  destination: uploadPath,
+  filename: (req, file, cb) => {
+    const uniqueSuffix = Date.now() + '-' + Math.round(Math.random() * 1e9);
+    cb(null, file.fieldname + '-' + uniqueSuffix);
+  },
+});
+
+const upload = multer({
+  storage,
+  limits: {
+    files: 4,
+    fileSize: 1024 * 1024 * 4,
+  },
+  fileFilter: (req, file, cb) => {
+    try {
+      if (expectedMimeTypes[file.mimetype]) {
+        cb(null, true);
+      } else
+        throw new UnprocessableEntity(
+          `Please use a valid image format. Valid formats include: ${Object.values(
+            VALID_IMAGE_FORMATS
+          ).join(', ')}`
+        );
+    } catch (error) {
+      cb(error);
+    }
+  },
+});
+
+module.exports = { upload };
```

Comment:
> @Olatisunkanmi: ~I might be a little naive but~ I think the middleware should be renamed from multer to media.
> 

I hear you. The reason it was named `multer.js` is because Multer supports various file types. The file exports the `upload` object that is eventually consumed at the routes where it will be applied. 

Do you think this addresses your concerns?

Reply:
I understand Tobi, but nonetheless. 
Think abstraction, if the file is name media. It would be a class to strictly handle all media uploads. 

The long term goal is for easy extension.